### ISSUE TITLE
Implementing abstract method using property declaration doesn't work well with accessibility check in constructor

### ISSUE THREAD

**Issue by ajafff** [Microsoft/TypeScript#26643]
**TypeScript Version:**  3.0

**Search Terms:**

**Code**

```ts
abstract class Base {
    abstract method(): void;
    constructor() {
        this.method();
    }
}

class Derived extends Base {
    method = () => {} // causes crash in constructor of base class
}
```

**Expected behavior:**

Maybe don't allow abstract methods to be implemented as property declaration?
I don't know if _abstract_ methods should be special cased. If the method in the base class was not abstract, there would still be an inconsistency. The constructor of the base class will use the prototype method. After the derived constructor assigned the property, every call site will use the instance property.

**Actual behavior:**

No indication this will blow up.

**Playground Link:** http://www.typescriptlang.org/play/#src=abstract%20class%20Base%20%7B%0A%20%20%20%20abstract%20method()%3A%20void%3B%0A%20%20%20%20constructor()%20%7B%0A%20%20%20%20%20%20%20%20this.method()%3B%0A%20%20%20%20%7D%0A%7D%0A%0Aclass%20Derived%20extends%20Base%20%7B%0A%20%20%20%20method%20%3D%20()%20%3D%3E%20%7B%7D%20%2F%2F%20causes%20crash%20in%20constructor%20of%20base%20class%0A%7D

**Related Issues:** #17113 requested the current behavior

**Comment by RyanCavanaugh** [comment]
😕

Won't Fix? It seems bad to take a breaking change on this one, and it's already the case that calling an abstract method from the constructor is extremely suspect since derived class properties won't be initialized yet.

**Comment by RyanCavanaugh** [comment]
My coworker suggests a "That Ship has Sailed" label
